src/lib/translation.ts

```diff
@@ -0,0 +1,109 @@
+import { TemplateFunction } from "./createAdapter";
+import { AdapterSettings, Answers, getDefaultAnswer } from "./questions";
+import { formatJsonString, translateText } from "./tools";
+
+export type Languages =
+	| "en"
+	| "de"
+	| "ru"
+	| "pt"
+	| "nl"
+	| "fr"
+	| "it"
+	| "es"
+	| "pl"
+	| "zh-cn";
+
+export type TranslatedTerm = { [lang in Languages]?: string };
+
+const titles: Record<Languages, string> = {
+	en: "Adapter settings for ${adapterName}",
+	de: "Adaptereinstellungen für ${adapterName}",
+	ru: "Настройки адаптера для ${adapterName}",
+	pt: "Configurações do adaptador para ${adapterName}",
+	nl: "Adapterinstellingen voor ${adapterName}",
+	fr: "Paramètres d'adaptateur pour ${adapterName}",
+	it: "Impostazioni dell'adattatore per ${adapterName}",
+	es: "Ajustes del adaptador para ${adapterName}",
+	pl: "Ustawienia adaptera dla ${adapterName}",
+	"zh-cn": "${adapterName}的适配器设置",
+};
+
+export async function getTranslatedSettingsForLanguage(
+	language: Languages,
+	answers: Answers,
+): Promise<Record<string, string>> {
+	const adapterSettings: AdapterSettings[] =
+		answers.adapterSettings || getDefaultAnswer("adapterSettings")!;
+	const translatedSettings: Record<string, string> = {};
+	translatedSettings[`${answers.adapterName} adapter settings`] = titles[
+		language
+	].replace("${adapterName}", answers.adapterName);
```

Comment:
is it possible that this placeholder appears multiple times? If so, the search string must be changed to a regex.